### ISSUE TITLE
Query param 'hasOwnProperty' crashes the app

### ISSUE THREAD

**Issue by Tem12** [a85/POSTMan-Chrome-Extension#8924]
**Describe the bug**
App crashes when calling `GET/POST` request with disabled `hasOwnProperty` parameter and at least one other enabled parameter

**To Reproduce**
1. Create new `GET/POST` request
2. Type valid URL for the request
3. Select `Params` and add query param with key `hasOwnProperty` and value `0`. (Value can be an empty or non-empty string)
4. Uncheck (disable) this param and create another parameter with key `a` and value `b` (Its key must be a non-empty string, the value can be again empty or non-empty string).
5. Send request
6. App crash

**Expected behavior**
The app should not crash when you have disabled param `hasOwnProperty` and even not sending it. It is probably caused by calling `hasOwnProperty` to check if property exists. But an already existing name `hasOwnProperty` cause to be called as a function which will lead to error.

**App information (please complete the following information):**
 - App Type: Native App
 - Postman Version: 7.30.1
 - OS: macOS Catalina 10.15.6

**Additional context**
Make sure you follow the order of adding parameters. 1st: `hasOwnProperty`, 2nd: `a`. Also if you forget to disable 1st param before adding 2nd then `Something went wrong` screen will show up. It will show at the moment when you try to type something in key input of 2nd param

**Comment by DannyDainton** [comment]
Hey @Tem12 

Thanks for writing in 🏆  - I was able to reproduce the issue following your steps, really appreciate the detail you added!!

For the `Something went wrong` error (which shouldn't be happening) I just opened an empty request and added the `hasOwnProperty` param key and as soon as I start writing the second param I saw the issue.

From View > Developer > Show DevTools (Current View) - I saw the following error:

```
TypeError: this.reference.hasOwnProperty is not a function
    at PostmanPropertyList.insert (vendor-shared.js:81900)
    at PostmanPropertyList.add (vendor-shared.js:81958)
    at arrayEach (vendor-shared.js:506221)
    at Function.forEach (vendor-shared.js:515073)
    at PostmanPropertyList.populate (vendor-shared.js:82074)
    at new PostmanPropertyList (vendor-shared.js:81832)
    at PostmanUrl.update (vendor-shared.js:108683)
    at new PostmanUrl (vendor-shared.js:108588)
    at getUrlParamString (vendor-shared.js:183277)
    at RequestUrlParamsEditor.setParams (vendor-shared.js:886817)
    at RequestUrlParamsEditor.handleParamsChange (vendor-shared.js:886837)
    at RequestURLParamsEditor.handleURLParamChange (vendor-shared.js:425595)
    at KeyValueEditor.handleValuesChange (vendor-shared.js:185965)
    at KeyValueFormEditor.handleValuesChange (vendor-shared.js:843014)
    at KeyValueFormEditor.handleChange (vendor-shared.js:842958)
    at KeyValueFormSortable.handleChange (vendor-shared.js:843221)
```

**Comment by Tem12** [comment]
I'm glad it helped

**Comment by DannyDainton** [comment]
It really did, we will update the thread when we have more information.

Thanks again 🚀

**Comment by sadakchap** [comment]
Hi @DannyDainton , can I work on it as my first issue ?

**Comment by codenirvana** [comment]
@sadakchap Feel free to raise a PR with the fix [here](https://github.com/postmanlabs/postman-collection). From the above logs, I see it's happening in [PostmanPropertyList](https://github.com/postmanlabs/postman-collection/blob/develop/lib/collection/property-list.js).

**Comment by mccannt** [comment]
This issue is now resolved in v7.32 of the application.
Closing this issue. If you are still experiencing this issue in the latest version, please let us know.
